### PR TITLE
boost-libs package

### PR DESCRIPTION
**Pull request by spahl**
High level description including:
 - use separate boost package instead of the 3rd party mesos included one
 - this is in preparation of merging metrics (https://github.com/dcos/dcos/pull/778)

# Checklist

 - [ ] Included a test which will fail if code is reverted but test is not
 - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

If this is a component/package update, include

 - [x] Change Log from last: New component
 - [ ] Test Results: n/a (this is the same lib, same version, just a different path)
 - [ ] Code Coverage: n/a

### CODE REVIEW

File: packages/mesos-modules/build [hunk 1]
```diff
@@ -7,9 +7,9 @@ popd
 source /opt/mesosphere/environment.export
 
 # Add the 3rdparty directories.
-export CPPFLAGS=-I/opt/mesosphere/lib/mesos/3rdparty/include
+export CPPFLAGS=-I /opt/mesosphere/active/boost-libs/include -I/opt/mesosphere/lib/mesos/3rdparty/include
```
Comment (malnick) [inline packages/mesos-modules/build]: I think this needs single quotes around the entire line.

Comment (spahl) [inline packages/mesos-modules/build]: good catch

**Comment by spahl** [comment]
This is ready for review

File: packages/boost-libs/build [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+# Copy all the headers.
+mkdir -p $PKG_PATH/include/
+cp -r boost $PKG_PATH/include/
+
+# Remove some bits of boost that even modules do not need.
+rm -rf $PKG_PATH/include/boost/phoenix $PKG_PATH/include/boost/fusion $PKG_PATH/include/boost/spirit
+
+# Build the following shared libraries.
+./b2 --with-filesystem --with-iostreams --with-program_options --with-system -s NO_BZIP2=1
```
Comment (cmaloney) [inline packages/boost-libs/build]: nit: would be nice to run `-j $NUM_CORES`

**Comment by amitaekbote** [comment]
🚢

**Pull request merged**